### ISSUE TITLE
Questions about using `mux`

### ISSUE THREAD

**Issue by Dreamacro** [shadowsocks/v2ray-plugin#33]
## Problem

The flow of the plugin:

```
DialA
          Client ==> Plugin (mux session) ==> Server (mux session)
DialB
```
The Client and Server connected with a persistent connection. In simple case, It's fine.

But when I have many server nodes, It becomes much complexity:

```
RequestA ==> Plugin (mux session1) ==> Server1 (mux session1)
RequestB ==> Plugin (mux session2) ==> Server2 (mux session2)
......
```

The cost of the mux is too expensive. BTW, because of the uncertainty of request address, I have to create a new sub connection.

In `simple-obfs`, a simpler and better relay method was used.

pseudocode:

```go
go io.Copy(remoteConn, localConn)
io.Copy(localConn, remoteConn)
```

## Question

Why does the new plugin deprecated the old relay mode and use the `mux` mode?

**Comment by VictoriaRaymond** [comment]
Mux is an optional feature AFAIK. You don't have to enable it if it doesn't fit your scenario.

**Comment by Dreamacro** [comment]
@VictoriaRaymond In `v2ray-plugin`, the `mux` is used by default. My project focuses on the use of the client.  So I can't decide what the server uses. When you have a large number of proxy nodes, I think using `mux` is `heavy`.

**Comment by VictoriaRaymond** [comment]
The default setting is mux = 1. It literally doesn't add any overhead to the system.

If you are writing an mux-compatible client, you can follow [its spec](https://v2ray.com/developer/protocols/muxcool.html).

**Comment by Dreamacro** [comment]
After I wrote a compatible implementation, I found that using the Websocket (without TLS) may have the risk of being detected.

Even if `Mux.Cool` is only a multiplex transmission protocol. I know this protocol is to have different sessions in the same TCP connection (like frp). But it's created by `Project V`. Will `Mux.Cool` be used as the basis for detection? Or is there another similar protocol?

**Comment by Mygod** [comment]
ws (without TLS) is meant to be used behind another HTTP server which is protected by TLS. Do not use ws directly (other than diagnosis/debugging purposes).

**Comment by Dreamacro** [comment]
@Mygod Please write this **note** in README.md

**Comment by Mygod** [comment]
Well then @VictoriaRaymond might jump out of nowhere and tell me it's against their philosophy of letting users decide everything. 😛 I think leaving my comment here would work just fine.

**Comment by Dreamacro** [comment]
I don't think writing a **note** in README.md contrary to the philosophy "letting users decide everything".

**Comment by VictoriaRaymond** [comment]
> Will `Mux.Cool` be used as the basis for detection? Or is there another similar protocol?

Yes. There are chances that 'mux' protocol get detected, as it doesn't include any encryption features. However, WebSocket has good enough obfuscation to hide the fingerprint. 

> it's against their philosophy of letting users decide everything.

It's up to you. But early warning: many users don't understand the un-balanced configuration, TLS prior to CDN and non-TLS after CDN. It will cause some trouble to explain such configuration.

**Comment by 2muaddlb** [comment]
Hi everyone!
Will there be any slicing of outgoing TCP-connections by V2Ray? I.e. I have satellite internet that shapes every single TCP-connection up to 5 Mbps, but if I run iPerf3 with 10 concurrent connections, I get 50 Mbps.
Since every program connects to sever via 1 TCP-connection, hence slow speed...
Muxing only reduces TCP handshake latency, the total throughput is not higher as per V2Ray docs
